File: pictures_and_rooms.py
def ppl_in_rooms(rooms,n):
    list_of_ppl = [1]*n
    non_empty_rooms = 0
    # prev_list = [0]*n
    how_many_options = set(rooms)
    m = n
    if len(how_many_options) != n: m = len(how_many_options)+1
    for _ in range(m):        
        for j in range(n):
            if list_of_ppl[j] > 0:
                list_of_ppl[j] -= 1
                list_of_ppl[rooms[j]-1] += 1
        # if prev_list == list_of_ppl: break
        # else: prev_list = list_of_ppl[:]

    for i in range(n): 
        if list_of_ppl[i] > 0: non_empty_rooms += 1
    return non_empty_rooms

def main():

    data_sets = int(input())
    results = [0]*data_sets
    for i in range(data_sets):
        n = int(input())
        rooms = list(map(int,input().split()))
        results[i] = ppl_in_rooms(rooms,n)
    for result in results:
        print(result)

main()

File: pictures_and_rooms.cpp
#include <iostream>
#include <vector>
 
using namespace std;
 
int ppl_in_rooms(vector<int>& rooms, int n) {
    vector<int> list_of_ppl(n, 1);
    int non_empty_rooms = 0;
    vector<int> prev_list;
 
    for (int i = 0; i < n; ++i) {
        for (int j = 0; j < n; ++j) {
            if (list_of_ppl[j] > 0) {
                list_of_ppl[j]--;
                list_of_ppl[rooms[j] - 1]++;
            }
        }
 
        if (prev_list == list_of_ppl) {
            break;
        } else {
            prev_list = list_of_ppl;
        }
    }
 
    for (int i = 0; i < n; ++i) {
        if (list_of_ppl[i] > 0) {
            non_empty_rooms++;
        }
    }
 
    return non_empty_rooms;
}
 
int main() {
    int data_sets;
    cin >> data_sets;
    vector<int> results;
 
    for (int i = 0; i < data_sets; ++i) {
        int n;
        cin >> n;
        vector<int> rooms(n);
 
        for (int j = 0; j < n; ++j) {
            cin >> rooms[j];
        }
 
        results.push_back(ppl_in_rooms(rooms, n));
    }
 
    for (int result : results) {
        cout << result << endl;
    }
 
    return 0;
}